gravitee-am-test/specs/gateway/login-flow.jest.spec.ts

```diff
@@ -0,0 +1,388 @@
+            "username": "user.five",
+            "email": secondCommonEmail,
+            "password": secondCommonPassword,
+            "client": multiUserLoginApp.id,
+            "source": customIdp.id,
+            "additionalInformation": {
+                "contract": contractValue
+            },
+            "preRegistration": false
+        });
+
+        user6 = await createUser(domain.id, accessToken, {
+            "firstName": "james",
+            "lastName": "hen",
+            "username": "user.six",
+            "email": secondCommonEmail,
+            "password": secondCommonPassword,
+            "client": multiUserLoginApp.id,
+            "source": customIdp.id,
+            "additionalInformation": {
+                "contract": contractValue
+            },
+            "preRegistration": false
+        });
+
+        expect(user3).toBeDefined();
+    });
+
+    it("all users should be able to login using username and password", async () => {
+        const clientId = multiUserLoginApp.settings.oauth.clientId;
+        const user1TokenResponse = await loginUserNameAndPassword(clientId, user1, user1Password);
+        expect(user1TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user1TokenResponse);
+
+        const user2TokenResponse = await loginUserNameAndPassword(clientId, user2, commonPassword);
+        expect(user2TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user2TokenResponse);
+
+        //user3 has same password as user2
+        const user3TokenResponse = await loginUserNameAndPassword(clientId, user3, commonPassword);
+        expect(user3TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user3TokenResponse);
+
+        //user3 has same password as user2
+        const user4TokenResponse = await loginUserNameAndPassword(clientId, user4, user4Password);
+        expect(user4TokenResponse.headers['location']).toContain('callback?code=');
+        await logoutUser(openIdConfiguration.end_session_endpoint, user4TokenResponse);
+    });
+
+    if(jdbc === 'jdbc') {
+        console.log("executing jdbc specific test")
+        it("jdbc: both users should be able to login using additional information (email) and password", async () => {
+            const clientId = multiUserLoginApp.settings.oauth.clientId;
+            //log in user3
+            const user1TokenResponse = await loginAdditionalInfoAndPassword(clientId, commonEmail, commonPassword);
+            expect(user1TokenResponse.headers['location']).toContain('callback?code=');
+            await logoutUser(openIdConfiguration.end_session_endpoint, user1TokenResponse);
+
+            //log in user4
+            const user2TokenResponse = await loginAdditionalInfoAndPassword(clientId, commonEmail, user4Password);
```

Comment:
maybe `user2TokenResponse` should be named `user4TokenResponse` if the `user4Password `is in used?